### ISSUE TITLE
[EPIC] Fountain syntax extension

### ISSUE THREAD

**Issue by shaedrich** [nyousefi/Fountain#50]
I thought, it might be a good idea, to collect all issues concerning syntax extension to have kind of an overview of where we stand on these issues (maybe @nyousefi can create a label for that):
| Title | Status | Proposed syntax | Rendered as | Suggested by |
| ----- | ------ | --------------- | ------------ | ----------- |
| [Stage directions](../20#issuecomment-587825855) | discussion | `+stage directions+` | `<div style="text-align:right;">stage directions</span>` | @Reaper10 |
| [Sound effects](../31) | discussion | <ul><li>`^sound effect`</li><li>`<sound effect`</li><li>`&sound effect`</li><li>`^sound effect^`</li></ul> | | @Reaper10 |
| [Scene metadata](../20) | discussion | |  | @severdia |
| [Element references](../31#issuecomment-1583440959) | discussion | `[{label}]({type}:{identifier})` | `<a href="[{type}:{identifier}">{label}</a>` utilizing [CURIE](https://talk.commonmark.org/t/compact-uris-curies/553/9)/[URN](https://en.wikipedia.org/wiki/Uniform_Resource_Name) | @shaedrich |
| [dramatis personae](../15) | discussion | table | https://scriptigo.com/help/fountain-format/133-improvements-to-the-fountain-format | @severdia |
| Script lining | discussion | `[[` and `]]`, starting with the word "LINE" | vertical lines with different colors | @shaedrich |
| Annotations | discussion | extended markdown syntax to [highlight text](https://www.markdownguide.org/extended-syntax#highlight) (`==` and `==`) in connection with [footnotes](https://www.ii.com/links-footnotes-markdown/#footnote-examples) (`[^1]` and `[^abbr]: endnote text` or `^[inline footnote]`) | `<mark title="{label}">{text selection}</mark>` | @shaedrich

**Comment by stovak** [comment]
I am +1 on the script lining.

I would, however prefer something like 

```
[[SHOT "Scene 3 shot 3"]]
Tom shot himself in the head and
[[/SHOT]]
[[SHOT "Scene 3 shot 4"]]
The gun fell to the floor.
[[/SHOT]]
```

Then you could run a production report and get a shot list.

**Comment by shaedrich** [comment]
Would you say, that a line and a shot are entirely identical?

**Comment by stovak** [comment]
> Would you say, that a line and a shot are entirely identical?

yes. My bad. Line is the visual implementation of a shot item. Either work, IMHO

**Comment by stovak** [comment]
```
[[LINE <label>]]
Tom shot himself in the head and
[[LINE <label2>]]
The gun fell to the floor.
[[/LINE]] [[/LINE]]
```

or you could do them like HTML tags... `<line label="blah"></line>`

**Comment by stovak** [comment]
If you implemented them as "html-style" tags, you could have a writing view that ignores production tags, and a production view that you can match up to storyboards.

```
<line label="blah" storyboard="storyboard.svg">
   Tom shot himself in the head and
     <line label="blah2" storyboard="storyboard2.svg">
        The gun fell to the floor.
     </line>
</line>

```

**Comment by shaedrich** [comment]
> > Would you say, that a line and a shot are entirely identical?
> 
> yes. My bad. Line is the visual implementation of a shot item. Either work, IMHO

@stovak Thank you for your suggestions, much appreciated 👍🏻  Sorry for my late reply. I needed to settle on how to put my understanding into words 😅 

My understanding was, that they are similar, yet not entirely identical, however, they could happen to point to the same thing, just not in general. What I mean, is that in my head (correct me if I'm wrong), a line is more synonymous with a camera setup than a shot since a camera setup _can_ consist of one shot only but _can_ also consist of multiple. But I'm not an expert, so I might be mistaken. Also, while line is a pre-production term, shot is a principal photography term and that might feel like mixing two very different workflows of two very different crew members.

What I like about your XML/HTML style tags is the ability to add additional attributes like the storyboards 👍🏻
